src/main/java/com/google/sps/travelbud/CountryServlet.java

```diff
@@ -35,12 +35,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
 
     String endpoint = request.getPathInfo();
+    String countryName = request.getParameter("name");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     Gson gson = new Gson();
-
-    // return the list of countries
-    if (endpoint == null) {
+    if (countryName != null) {
+      List<Country> filteredCountries = Country.filter(datastore, countryName);
+      response.getWriter().println(gson.toJson(filteredCountries));
+    } else if (endpoint == null) {
       List<Country> countries = Country.getAll(datastore);
```

Comment:
Suggestion: don't overcomplicate
```suggestion
if (endpoint == null) {
  return Country.getAll(datastore).stream()
      .filter(c -> countryName == null || c.name.toLowerCase().contains(countryName.toLowerCase()))
      .collect(Collectors.toList());
```

You can delete the filtering functions from the Country class.